stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/view/EndlessMessageListScrollListener.kt

```diff
@@ -59,20 +71,61 @@ public class EndlessMessageListScrollListener(
             throw IllegalStateException("EndlessScrollListener supports only LinearLayoutManager")
         }
 
-        checkScrollUp(dy, layoutManager, recyclerView)
+        handleScroll(dy, layoutManager, recyclerView)
     }
 
     /**
      * Checks if the scroll is going up and if the threshold number of items has been shown. If the scroll is downwards,
      * then it stops the check.
      */
-    private fun checkScrollUp(dy: Int, layoutManager: LinearLayoutManager, recyclerView: RecyclerView) {
-        if (dy >= 0) {
-            // Scrolling downwards
-            return
+    private fun handleScroll(dy: Int, layoutManager: LinearLayoutManager, recyclerView: RecyclerView) {
+        when {
+            dy >= 0 && shouldFetchBottomMessages && firstMessageBellowGapPosition != null -> {
+                handleScrollDown(layoutManager, recyclerView, firstMessageBellowGapPosition!!)
+            }
+
+            dy < 0 -> {
+                handleScrollUp(layoutManager, recyclerView)
+            }
+        }
+    }
+
+    private fun canRequestNow(): Boolean {
+        return (Date().time - lastRequest > bottomRequestTriggerDelay)
```

Comment:
shouldn't the check be if loading is in progress return false, otherwise true, why are we using a delay?